### PR TITLE
Calibration model training script.

### PR DESCRIPTION
**Pull request by ayush-1506**
To do:

- Tests. Verified locally but need to write tests
- Expose this to CLI

### CODE REVIEW

File: dialogy/plugins/text/calibration/xgb.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        return np.array(features), targets
+
+
+class CalibrationModel(Plugin):
+    def __init__(self) -> None:
+        self.extraction_pipeline = FeatureExtractor()
+        self.clf = XGBRegressor(n_jobs=4)
+
+    def train(self, filename: str, model_name: str) -> None:
```
Comment (ltbringer) [inline dialogy/plugins/text/calibration/xgb.py:92]: Let's get the dataframe here. As an example, refer to [this line](https://github.com/skit-ai/dialogy/blob/9137e86197ec6f7d015b6c7c00945d7c11115e73/dialogy/plugins/text/classification/xlmr.py#L197). It is easier to provide small mock dataframes to plugins to test a few cases.

File: dialogy/plugins/text/calibration/xgb.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        return self.clf.predict(self.extraction_pipeline.features(alternatives))
+
+    def save(self, fname: str) -> None:
+        pickle.dump(self, open(fname, "wb"))
+
+    def utility(self, *args: Any) -> Any:
+        return self.inference(*args)  # pylint: disable=no-value-for-parameter
+
+
+if __name__ == "__main__":
```
Comment (ltbringer) [inline dialogy/plugins/text/calibration/xgb.py:107]: I believe this was required for testing locally and not needed here?

File: dialogy/plugins/text/calibration/xgb.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            alt["lm_score"] / l,
+            alt["transcript"].count("UNK"),
+            l,
+            alt["am_score"] / math.log(1 + l),
+            alt["lm_score"] / math.log(1 + l),
+            alt["am_score"] / math.sqrt(l),
+            alt["am_score"] / math.sqrt(l),
+        ]
+
+    def transform(self, filename: str) -> Tuple[np.ndarray, List]:
```
Comment (ltbringer) [inline dialogy/plugins/text/calibration/xgb.py:67]: Let's get the dataframe here. As an example, refer to [this line](https://github.com/skit-ai/dialogy/blob/9137e86197ec6f7d015b6c7c00945d7c11115e73/dialogy/plugins/text/merge_asr_output/__init__.py#L105). It is easier to provide small mock dataframes to plugins to test a few cases.

**Review by ltbringer — CHANGES_REQUESTED** [review]
A comment on the plugin in general. We are not filtering the ASR alternatives here. Where would that happen if not here? Can we utilize the filters and thresholds to filter here itself?

Should that be the case, then can the plugin also have a transform method to do the same?
[This](https://github.com/skit-ai/dialogy/blob/9137e86197ec6f7d015b6c7c00945d7c11115e73/dialogy/workflow/workflow.py#L190) particular section ensures that once a plugin is trained, it can be used by other plugins in the sequence.

If applied to our case, the classifier would never see poor transcripts.

**Comment by ayush-1506** [comment]
This is for training our calibration model on transcripts, we don't remove alternatives here. Removing bad alternatives before training slu was added [here](https://github.com/skit-ai/dialogy-template-simple-transformers/pull/47/files#diff-b8c03ee7d6f475174f2fef2b93f4b9c3b059a4d0eff1eb15ab43e129904ac8e5R186), although it looks like this has been moved.

File: docs_src/conf.py [hunk 4]
```diff
@@ -13,10 +13,11 @@
 import os
 import sys
 import toml
-sys.path.insert(0, os.path.abspath('.'))
 
+sys.path.insert(0, os.path.abspath("."))
 
-with open(os.path.join(os.path.abspath('.'), "..", "pyproject.toml")) as f:
+
```
Comment (ayush-1506) [inline docs_src/conf.py:19]: `black` changes

File: pyproject.toml [hunk 5]
```diff
@@ -20,6 +20,8 @@ pandas = "^1.3.2"
 sklearn = "^0.0"
```
Comment (ayush-1506) [inline pyproject.toml:20]: `scipy` is used for a single test, might try to remove this dependency.

**Comment by ayush-1506** [comment]
Is it possible to remove `pickle` warning from codacy test? While there are general security issues with using pickle, I'm not sure if those are particularly relevant for us.

**Comment by ayush-1506** [comment]
Is it possible to remove `pickle` warning from codacy test? While there are general security issues with pickle, I'm not sure if those are particularly relevant for us.

**Pull request merged**